From bc073960643fb2c899bacdd3f2400aff5fe8689e Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 16 Nov 2023 16:38:47 +0100
Subject: [PATCH] drm/stm: ltdc: remove reload interrupt

The reload interrupt is not use by the driver. To avoid unnecessary call
of interrupt routine, don't enable it.

Change-Id: I6a9cbcc7c2566e08f9d2b053934bbf8852c09836
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -786,7 +786,7 @@ static void ltdc_crtc_atomic_enable(stru
 	regmap_write(ldev->regmap, LTDC_BCCR, BCCR_BCBLACK);
 
 	/* Enable IRQ */
-	regmap_set_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE | IER_RRIE | IER_TERRIE);
+	regmap_set_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE | IER_TERRIE);
 
 	/* Commit shadow registers = update planes at next vblank */
 	if (!ldev->caps.plane_reg_shadow)
@@ -811,7 +811,7 @@ static void ltdc_crtc_atomic_disable(str
 		regmap_write_bits(ldev->regmap, LTDC_L1CR + layer_index * LAY_OFS, LXCR_MASK, 0);
 
 	/* disable IRQ */
-	regmap_clear_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE | IER_RRIE | IER_TERRIE);
+	regmap_clear_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE | IER_TERRIE);
 
 	/* immediately commit disable of layers before switching off LTDC */
 	if (!ldev->caps.plane_reg_shadow)
@@ -1994,10 +1994,10 @@ int ltdc_load(struct drm_device *ddev)
 
 	/* Disable interrupts */
 	if (ldev->caps.fifo_threshold)
-		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_RRIE | IER_FUWIE |
+		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_FUWIE |
 				  IER_TERRIE);
 	else
-		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_RRIE | IER_FUWIE |
+		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_FUWIE |
 				  IER_TERRIE | IER_FUEIE);
 
 	DRM_DEBUG_DRIVER("ltdc hw version 0x%08x\n", ldev->caps.hw_version);
